### ISSUE TITLE
Pie/donut chart slice rectangle focus covering the slice and more in some cases

### ISSUE THREAD

**Issue by hakansemerci** [highcharts/highcharts#18668]
**Live demo with steps to reproduce**
https://codepen.io/anon/pen/dRZLNo

1. Click in the render area to give focus to the iframe
2. Tab through the chart

**Expected behaviour**
The focus should be clearly identified by highlighting the border of the focused slice. The slice selected should have a focus highlight around the slice border covering the specific slice only.

**Actual behaviour**
The focus is rectangular, and it is hard to identify the specific slice when the slices are small.
![Screenshot 2023-03-14 at 16 19 53](https://user-images.githubusercontent.com/31401102/225050502-007cf175-eb74-457f-84af-6b83455142ec.png)

The halo and border option could be used but it is not good when the pattern is one. Also, the border can not surround the whole slice in some cases.
https://jsfiddle.net/8n7orvaz/
![Screenshot 2023-03-14 at 16 33 41](https://user-images.githubusercontent.com/31401102/225053528-e2517204-92b3-4a94-b535-409d10071933.png)

**Comment by karolkolodziej** [comment]
Hi @hakansemerci!
Thank you for describing this feature.

FYI @oysteinmoseng
